docs/advanced/pool.md

```diff
@@ -0,0 +1,88 @@
+# Custom Pool
+
+::: warning
+This is advanced API. If you are just running tests, you probably don't need this. It is primarily used by library authors.
+:::
+
+Vitest runs tests in pools. By default, there are several pools:
+
+- `threads` to run tests using `node:worker_threads` (isolation is provided with a new worker context)
+- `forks` to run tests using `node:child_process` (isolation is provided with a new `child_process.fork` process)
+- `vmThreafs` to run tests using `node:worker_threads` (but isolation is provided with `vm` module instead of a new worker context)
+- `browser` to run tests using browser providers
+- `typescript` to run typechecking on tests
+
+You can provide your own pool by specifying a file path:
+
+```ts
+export default defineConfig({
+  test: {
+    // will run every file with a custom pool by default
+    pool: './my-custom-pool.ts',
```

Comment:
Would this also allow specifying names of published npm packages? If so, when ever we introduce new built-in pool, it would be considered as breaking change. Imagine a situation where someone publishes `vmForks` custom pool into npm before we implement that as built-in.

Reply:
Yes, it would be a breaking change then.

Reply:
We could require the `pool` option to be prefixed with `vitest-pool-` if it's not path. Then all non-fixed would be preserved for built-in ones.

Reply:
Looks like the same problem applies to custom reporters so I guess it's not an issue really.